python/paddle/amp/auto_cast.py

```diff
@@ -168,15 +213,15 @@ def _is_xpu_bfloat16_supported():
     )
 
 
-def _is_custom_device_bfloat16_supported():
+def _is_custom_device_bfloat16_supported() -> bool:
     """
     Judge whether current custom device support bfloat16 amp.
     """
     place = _current_expected_place()
     return place.get_device_type() == 'npu'
 
 
-def need_keep_fp32(layer, dtype):
+def need_keep_fp32(layer: _LayerList, dtype: dtype) -> bool:
```

Comment:
```suggestion
def need_keep_fp32(layer: _LayerList, dtype: str) -> bool:
```

后面

``` python
    elif (layer._dtype == 'float16') or (
        (dtype == 'float16')
```

做字符串比对，应该只能是 str ～